index.js

```diff
@@ -29,6 +29,9 @@ module.exports = fp(function from (fastify, opts, next) {
     sessionTimeout: opts.sessionTimeout,
     undici: opts.undici
   })
+
+  fastify.setErrorHandler((error, request, reply) => errorHandler(error, request, reply, onErrorDefault))
```

Comment:
unfortunately this is not viable either. This will be implemented by the developers using the plugin.

Reply:
are you saying that `fastify.setErrorHandler` may be used higher up in the context and this would override it?

Perhaps there is a way we can intercept and change the error but still pass it up to the parent context error handler?

Reply:
I don't think we should intercept that error.

Reply:
adding an error handler here will override any other that were previously set.

Reply:
It would be nice for the user to get a 504 rather than 500, but I agree it may be overkill to handle this at the moment.

I've reverted the last few commits, removed the redundant check for UND_ERR_BODY_TIMEOUT and added a test.